core/drand_beacon.go

```diff
@@ -360,6 +356,36 @@ func (bp *BeaconProcess) isFreshRun() bool {
 	return errG != nil || errS != nil
 }
 
+// getChainHash return the beaconID of that beaconProcess, if set
+func (bp *BeaconProcess) getBeaconID() string {
+	return bp.beaconID
+}
+
+// getChainHash return the HashChain in hex format as a string
+func (bp *BeaconProcess) getChainHash() string {
+	bp.state.Lock()
+	defer bp.state.Unlock()
+	if bp.group != nil {
+		info := chain.NewChainInfo(bp.group)
+		return info.HashString()
+	}
+
+	return ""
+}
+
+func (bp *BeaconProcess) newMetadata() *common.Metadata {
+	metadata := common.NewMetadata(bp.version.ToProto())
+	metadata.BeaconID = bp.getBeaconID()
+
+	if hash, err := hex.DecodeString(bp.getChainHash()); err == nil {
```

Comment:
Given this will be called _for each outgoing network packet_ it may be worth just saving the chain hash directly in a field, instead of reconstructing it from the group and calling the hash function each time.